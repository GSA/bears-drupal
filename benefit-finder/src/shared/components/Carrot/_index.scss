#benefit-finder {
  .carrot {
    position: absolute;
    top: 0;
    bottom: 0;
    right: 20px;
    margin: auto 0;
    transform: rotate(-90deg);

    &::after {
      content: '\25B8';
    }
  }
}
