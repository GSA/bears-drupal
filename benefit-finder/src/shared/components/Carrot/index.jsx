import PropTypes from 'prop-types'
import './_index.scss'

/**
 * a functional component that renders an svg image
 * @component
 * @return {html} returns a semantic svg element
 */
const Carrot = ({ color }) => {
  const svg = (
    <svg
      className="carrot"
      xmlns="http://www.w3.org/2000/svg"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      fill={color}
    >
      <path d="M0 0h24v24H0z" fill="none" />
      <path d="M7 10l5 5 5-5z" />
    </svg>
  )

  return svg
}

Carrot.propTypes = {
  color: PropTypes.string,
}

export default Carrot
