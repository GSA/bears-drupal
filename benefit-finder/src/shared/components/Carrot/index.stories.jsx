import Carrot from './index.jsx'

export default {
  component: Carrot,
  tags: ['autodocs'],
  args: {
    color: '#005EA2',
  },
}

export const Primary = {}
