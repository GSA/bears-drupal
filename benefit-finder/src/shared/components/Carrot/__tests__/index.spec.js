import renderer from 'react-test-renderer'
import Carrot from '../index.jsx'

describe('Carrot', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<Carrot />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
