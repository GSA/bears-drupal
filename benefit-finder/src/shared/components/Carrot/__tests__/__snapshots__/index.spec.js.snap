// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrot renders a match to the previous snapshot 1`] = `
<svg
  className="carrot"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M7 10l5 5 5-5z"
  />
</svg>
`;
